File: src/main/java/io/vertigo/shiny/components/text/textpath/ShinyTextPath.java
package io.vertigo.shiny.components.text.textpath;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyTextPath(
		String path,
		String separator) implements ShinyComponent {

	@Override
	public String type() {
		return "ShinyTextPath";
	}

	public ShinyTextPath {
		Assertion.check().isNotBlank(path, "Path cannot be blank");
	}
}

File: banshee/src/models/ShinyTextPath.ts
import { ShinyComponent } from '../ShinyComponent';

export class ShinyTextPath implements ShinyComponent {
  path: string;
  separator?: string;
  type: string = 'textPath';

  constructor(path: string, separator?: string) {
    this.path = path;
    this.separator = separator;
  }
}
